easybuild/framework/easyconfig/tools.py

```diff
@@ -383,8 +384,12 @@ def stats_to_str(stats):
 
     txt = "{\n"
     pref = "    "
-    for (k, v) in stats.items():
-        txt += "%s%s: %s,\n" % (pref, quote_str(k), quote_str(v))
+    for key in sorted(stats):
+        if isyeb:
+            key, val = quote_yaml_special_chars(key), quote_yaml_special_chars(stats[key])
+        else:
+            key, val = quote_str(key), quote_str(stats[key])
+        txt += "%s%s: %s,\n" % (pref, key, val)
     txt += "}"
```

Comment:
hmm, does the format with `{...}` also work in `.yeb`?

i.e., what type of value does `buildstats` parameter have in the dumped & parsed easyconfig? Can you check that in the test you added?